### PR TITLE
Use NSArray/NSMutableArray and NSSet/NSMutableSet in NSOrderedSet

### PR DESCRIPTION
**Pull request by amorde**
Taking another stab at #1367 

This cleans up a lot of the implementation since we can defer work to `NS{Mutable}Array` and `NS{Mutable}Set`

### CODE REVIEW

File: Foundation/NSOrderedSet.swift [hunk 1]
```diff
@@ -252,22 +228,16 @@ extension NSOrderedSet {
     }
     
     public func objectEnumerator() -> NSEnumerator {
-        guard type(of: self) === NSOrderedSet.self || type(of: self) === NSMutableOrderedSet.self else {
-            NSRequiresConcreteImplementation()
-        }
-        return NSGeneratorEnumerator(_orderedStorage.map { __SwiftValue.fetch(nonOptional: $0) }.makeIterator())
+        return _orderedStorage.objectEnumerator()
```
Comment (amorde) [inline Foundation/NSOrderedSet.swift]: This has the interesting side effect of making these enumerators behave similar to Darwin - they will now break if mutation occurs during iteration. The previous implementation made a copy of the all the objects using `map` and therefore did not have this issue.

File: TestFoundation/TestNSOrderedSet.swift [hunk 2]
```diff
@@ -62,6 +61,126 @@ class TestNSOrderedSet : XCTestCase {
             index += 1
         }
     }
+    
+    func test_enumerationUsingBlock() {
```
Comment (amorde) [inline TestFoundation/TestNSOrderedSet.swift]: This test was copied from `NSArray` tests for coverage on concurrent enumeration

File: Foundation/NSOrderedSet.swift [hunk 3]
```diff
@@ -544,7 +581,8 @@ extension NSMutableOrderedSet {
     }
     
     open func intersectSet(_ other: Set<AnyHashable>) {
-        for case let item as AnyHashable in self where !other.contains(item) {
+        let objects = Array(self)
+        for case let item as AnyHashable in objects where !other.contains(item) {
```
Comment (amorde) [inline Foundation/NSOrderedSet.swift]: since the enumerator now breaks if mutation occurs, we have to make a copy here

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**